pkg/controller/elasticsearch/stackmon/validations.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package stackmon
+
+import (
+	"fmt"
+
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/version"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+)
+
+const (
+	unsupportedVersionForStackMonitoringMsg    = "Unsupported version for Stack Monitoring. Required >= %s."
+	invalidStackMonitoringElasticsearchRefsMsg = "Only one Elasticsearch reference is supported for %s Stack Monitoring"
+)
+
+var (
+	// Minimum Stack version to enable Stack Monitoring.
+	// This requirement comes from the fact that we configure Elasticsearch to write logs to disk for Filebeat
+	// via the env var ES_LOG_STYLE available from this version.
+	MinStackVersion = version.MustParse("7.14.0")
+)
+
+// Validate validates that the Elasticsearch version is supported for Stack Monitoring and that there is exactly one
+// Elasticsearch reference defined when Stack Monitoring is defined
+func Validate(es esv1.Elasticsearch) field.ErrorList {
```

Comment:
as a general principle I think any validation function should be unit-tested, even though they may be very simple functions? At least to catch future regressions!

Reply:
It's a good general principle, you're right, I'm adding unit tests.